package/kedro_viz/api/graphql.py

```diff
@@ -41,112 +42,113 @@
+    """Get a run by id from the session store.
+
+    Args:
+        run_id: ID of the run to fetch
 
     Returns:
-        list of Run objects
+        Run object
     """
-    return [get_run(ID("123"))]
-
-
-@strawberry.type
-class RunModelGraphQLType:
-    """RunModel format to return to the frontend"""
-
-    id: str
-    blob: str
+    session = data_access_manager.db_session
+    if not session:
+        return None
+    run_data = session.query(RunModel).filter(RunModel.id == run_id).first()
+    if not run_data:
+        logger.warning("Cannot find matching run")
+        return None
+    return format_run(run_data.id, json.loads(run_data.blob))
 
 
-def get_all_runs() -> typing.List[RunModelGraphQLType]:
-    """Gets all runs from the session store
+def get_all_runs() -> List[Run]:
+    """Get all runs from the session store.
 
     Returns:
         list of Run objects
-
     """
-    return [
-        RunModelGraphQLType(id=kedro_session.id, blob=kedro_session.blob)
-        for kedro_session in data_access_manager.db_session.query(RunModel).all()
-    ]
+    runs = []
+    session = data_access_manager.db_session
+    if not session:
+        return runs
+    for run_data in session.query(RunModel).all():
+        run = format_run(run_data.id, json.loads(run_data.blob))
+        runs.append(run)
+    return runs
 
 
 @strawberry.type
 class Run:
-    """Run object format to return to the frontend"""
+    """Run object format"""
 
     id: ID
-    bookmark: bool
-    timestamp: str
     title: str
-    metadata: RunMetadata
-    details: RunDetails
-
-
-@strawberry.type
-class RunMetadata:
-    """RunMetadata object format"""
-
-    id: ID
-    author: str
-    gitBranch: str
-    gitSha: str
-    notes: str
-    runCommand: str
+    timestamp: str
+    author: Optional[str]
+    gitBranch: Optional[str]
+    gitSha: Optional[str]
+    bookmark: Optional[bool]
+    notes: Optional[str]
+    runCommand: Optional[str]
 
 
 @strawberry.type
-class RunDetails:
-    """RunDetails object format"""
-
-    id: ID
-    name: str
-    details: str
-
+class TrackingDataSet:
+    """TrackingDataSet object to structure tracking data for a Run."""
 
+    datasetName: str
+    datasetType: str
+    data: JSONScalar
 @strawberry.type
 class Query:
     """Query endpoint to get data from the session store"""
 
-    @strawberry.field
-    def run(self, run_id: ID) -> Run:
-        """Query to get data for a specific run from the session store"""
-        return get_run(run_id)
+    runs_list: List[Run] = strawberry.field(resolver=get_all_runs)
 
-    runs: List[Run] = strawberry.field(resolver=get_runs)
-    all_runs: typing.List[RunModelGraphQLType] = strawberry.field(resolver=get_all_runs)
+    @strawberry.field
+    def run_metadata(self, run_ids: List[ID]) -> List[Run]:
+        """Query to get data for specific runs from the session store"""
+        runs = []
+        for run_id in run_ids:
+            run = get_run(run_id)
```

Comment:
Okay this is a useful thing to remember: when we write ORM code, we need to be mindful of how many queries we are issuing again the database. In this case, every run_id will trigger a `get_run` which is 1 db query. It's much better to select multiple run ids with a single query from the db using [filter](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.filter) instead.

I'd personally change `get_run` into `get_runs(run_ids: List[ID])` for this purpose.